pkg/virt-controller/watch/vm_test.go

```diff
@@ -1186,6 +1187,334 @@ var _ = Describe("VirtualMachine", func() {
 				})
 			})
 		})
+
+		Context("VM printableStatus", func() {
+
+			table.DescribeTable("should set a Stopped status when VMI doesn't exist", func(running bool, runStrategy v1.VirtualMachineRunStrategy) {
+				vm, _ := DefaultVirtualMachine(running)
+
+				if runStrategy != v1.RunStrategyUnknown {
+					vm.Spec.RunStrategy = &runStrategy
+					vm.Spec.Running = nil
+				}
+
+				addVirtualMachine(vm)
+
+				vmInterface.EXPECT().UpdateStatus(gomock.Any()).Times(1).Do(func(obj interface{}) {
+					objVM := obj.(*v1.VirtualMachine)
+					Expect(objVM.Status.PrintableStatus).To(Equal(v1.VirtualMachineStatusStopped))
+				})
+
+				controller.Execute()
+			},
+
+				table.Entry("running: false", false, v1.RunStrategyUnknown),
+				table.Entry("runStrategy: Halted", false, v1.RunStrategyHalted),
+				table.Entry("running: Manual", false, v1.RunStrategyManual),
+			)
+
+			table.DescribeTable("should set a Stopped status when VMI has stopped", func(phase v1.VirtualMachineInstancePhase, deletionTimestamp bool) {
+				vm, vmi := DefaultVirtualMachine(true)
+
+				vmi.Status.Phase = phase
+				if deletionTimestamp {
+					vmi.ObjectMeta.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+				}
+				addVirtualMachine(vm)
+				vmiFeeder.Add(vmi)
+
+				vmiInterface.EXPECT().Delete(gomock.Any(), gomock.Any()).Times(1)
+				vmInterface.EXPECT().UpdateStatus(gomock.Any()).Times(1).Do(func(obj interface{}) {
+					objVM := obj.(*v1.VirtualMachine)
+					Expect(objVM.Status.PrintableStatus).To(Equal(v1.VirtualMachineStatusStopped))
+				})
+
+				controller.Execute()
+			},
+
+				table.Entry("in Succeeded state", v1.Succeeded, false),
+				table.Entry("in Succeeded state with a deletionTimestamp", v1.Succeeded, true),
+				table.Entry("in Failed state", v1.Failed, false),
+				table.Entry("in Failed state with a deletionTimestamp", v1.Failed, true),
+			)
+
+			table.DescribeTable("should set a Provisioning status when VMI doesn't exist", func(running bool, runStrategy v1.VirtualMachineRunStrategy) {
+				vm, vmi := DefaultVirtualMachine(running)
+
+				if runStrategy != v1.RunStrategyUnknown {
+					vm.Spec.RunStrategy = &runStrategy
+					vm.Spec.Running = nil
+				}
+
+				addVirtualMachine(vm)
+
+				vmiInterface.EXPECT().Create(gomock.Any()).Return(vmi, nil)
+
+				vmInterface.EXPECT().UpdateStatus(gomock.Any()).Times(1).Do(func(obj interface{}) {
+					objVM := obj.(*v1.VirtualMachine)
+					Expect(objVM.Status.PrintableStatus).To(Equal(v1.VirtualMachineStatusProvisioning))
+				})
+
+				controller.Execute()
+			},
+
+				table.Entry("running: true", true, v1.RunStrategyUnknown),
+				table.Entry("runStrategy: Always", true, v1.RunStrategyAlways),
+				table.Entry("runStrategy: RerunOnFailure", true, v1.RunStrategyRerunOnFailure),
```

Comment:
I can't fail to notice that there's an input parameter for `Running` but you're passing it `true` in all test entries; why bother ? Couldn't you just omit the parameter and assume it'll always be true ?

Reply:
Well, I failed to notice that :-)
Will remove.